coordination_network_toolkit/preprocess.py

```diff
@@ -46,44 +46,46 @@ def preprocess_data(db_path: str, messages: Iterable):
 
     db = initialise_db(db_path)
 
-    db.execute("begin")
-    processed = (
-        (
-            message_id,
-            user_id,
-            username,
-            repost_id or None,
-            reply_id or None,
-            message,
-            len(message),
-            zlib.adler32(message.encode("utf8")),
-            # This will be populated only when similarity calculations are necessary
-            None,
-            int(timestamp),
-            urls.split(" ") if urls else [],
+    try:
+        db.execute("begin")
+        processed = (
+            (
+                message_id,
+                user_id,
+                username,
+                repost_id or None,
+                reply_id or None,
+                message,
+                len(message),
+                zlib.adler32(message.encode("utf8")),
+                # This will be populated only when similarity calculations are necessary
+                None,
+                float(timestamp),
+                urls.split(" ") if urls else [],
+            )
+            for message_id, user_id, username, repost_id, reply_id, message, timestamp, urls in messages
         )
-        for message_id, user_id, username, repost_id, reply_id, message, timestamp, urls in messages
-    )
 
-    for row in processed:
-        db.execute(
-            "insert or ignore into edge values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)",
-            row[:-1],
-        )
+        for row in processed:
+            db.execute(
+                "insert or ignore into edge values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)",
+                row[:-1],
+            )
 
-        message_id, user_id = row[:2]
-        timestamp = row[-2]
+            message_id, user_id = row[:2]
+            timestamp = row[-2]
 
-        # Ignore url shared in reposts
-        if not row[3]:
-            for url in row[-1]:
-                db.execute(
-                    "insert or ignore into message_url values(?, ?, ?, ?)",
-                    (message_id, url, timestamp, user_id),
-                )
+            # Ignore url shared in reposts
+            if not row[3]:
+                for url in row[-1]:
+                    db.execute(
+                        "insert or ignore into message_url values(?, ?, ?, ?)",
+                        (message_id, url, timestamp, user_id),
+                    )
 
-    db.execute("commit")
-    db.close()
+        db.execute("commit")
+    finally:
+        db.close()
```

Comment:
Good catch, I always forget to do this if I don't have a proper context manager for transactions.

Reply:
I may have had problems parsing my example csv for the library usage example, resulting in db connections left open, and that's how I noticed... 😸